fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/jobs/applyholidaystoloans/ApplyHolidaysToLoansTasklet.java

```diff
@@ -129,13 +131,13 @@ private void adjustRepaymentSchedules(Loan loan, Holiday holiday, LocalDate adju
                 loanRepaymentScheduleInstallment.updateFromDate(tmpFromDate);
             }
 
-            if (!DateUtils.isBefore(oldDueDate, holiday.getFromDate())) {
+            if (isDateWithinRange(oldDueDate, holiday.getFromDate(), holiday.getToDate())) {
                 // FIXME: AA do we need to apply non-working days.
                 // Assuming holiday's repayment reschedule to date cannot be
                 // created on a non-working day.
 
-                adjustedRescheduleToDate = scheduledDateGenerator.generateNextRepaymentDate(adjustedRescheduleToDate, loanApplicationTerms,
-                        false);
+                adjustedRescheduleToDate = scheduledDateGenerator.generateNextRepaymentDateWhenHolidayApply(adjustedRescheduleToDate,
+                        loanApplicationTerms);
```

Comment:
Can you please fix an another issue?
Line 90: holiday.isProcessed() -> holiday.setProcessed(true);